sensor/kubernetes/upgrade/deployment.go

```diff
@@ -40,6 +41,12 @@ func toK8sEnvVars(triggerEnvVars []*central.SensorUpgradeTrigger_EnvVarDef) []v1
 }
 
 func (p *process) determineImage() (string, error) {
+	log.Infof("Sensor was instructed to run the upgrader with image: %s", p.trigger.GetImage())
+	// TODO: check the pullability of that image
+	if env.UpgraderImageOverwrite.Setting() != "" {
+		log.Infof("Sensor will use the follownig upgrader image instead: %s", env.UpgraderImageOverwrite.Setting())
```

Comment:
typo
```suggestion
		log.Infof("Sensor will use the following upgrader image instead: %s", env.UpgraderImageOverwrite.Setting())
```

And aside from that a design question, why do we need to have an `UpgraderImageOverwrite`? Is it just for debugging / is it hard to set the image in `p.trigger` or is this used somewhere in production too? I find it a bit strange that the image can be set in two places like that.

Reply:
> And aside from that a design question, why do we need to have an UpgraderImageOverwrite? Is it just for debugging / is it hard to set the image in p.trigger or is this used somewhere in production too? I find it a bit strange that the image can be set in two places like that.

The commit [742d10d](https://github.com/stackrox/stackrox/pull/12436/commits/742d10d40b2b66ee29056726c61434eb0ca1ac58) needs to be reverted. I just left it there in case any reviewers wanted to test it themselves.